pkg/certificate/certificate.go

```diff
@@ -276,7 +276,9 @@ func GenerateRequest(request *Request, privateKey crypto.Signer) error {
 func (request *Request) GenerateCSR() error {
 	certificateRequest := x509.CertificateRequest{}
 	certificateRequest.Subject = request.Subject
-	certificateRequest.DNSNames = request.DNSNames
+	if !request.OmitSANs {
+		certificateRequest.DNSNames = request.DNSNames
+	}
 	certificateRequest.EmailAddresses = request.EmailAddresses
 	certificateRequest.IPAddresses = request.IPAddresses
 	certificateRequest.URIs = request.URIs
```

Comment:
@mr-tron do you think we should be ignoring all SAN types since the option is general `--omit-sans`?  It won't matter for the customer issue because in general public CAs only support DNS SANs but I'm thinking it is probably best for the option to apply to all SAN types we support.

Reply:
so you suggest to move emails, IPs and URIs inside `if` condition?